pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -9,8 +9,65 @@ let
     # to build it. This is a bit confusing for cross compilation.
     inherit (stdenv) hostPlatform;
   };
+
+  makeOverlayable = mkDerivationSimple: # TODO(@robert): turn mkDerivationSimple into let binding.
+    fnOrAttrs:
+      if builtins.isFunction fnOrAttrs
+      then makeDerivationExtensible mkDerivationSimple fnOrAttrs
+      else makeDerivationExtensible0 mkDerivationSimple fnOrAttrs;
+
+  # Based off lib.makeExtensible, with modifications:
+  makeDerivationExtensible = mkDerivationSimple: rattrs:
+    let
+      args = rattrs (args // { inherit public; });
+      public =
+        mkDerivationSimple
+          (f0:
+            let
+              f = self: super:
+                # Convert f0 to an overlay. Legacy is:
+                #   overrideAttrs (super: {})
+                # We want to introduce self. We follow the convention of overlays:
+                #   overrideAttrs (self: super: {})
+                # Which means the first parameter can be either self or super.
+                # This is surprising, but far better than the confusion that would
+                # arise from flipping an overlay's parameters in some cases.
+                let x = f0 super;
+                in
+                  if builtins.isFunction x
+                  then
+                    # Can't reuse `x`, because `self` comes first.
+                    # Looks inefficient, but `f0 super` was a cheap thunk.
+                    f0 self super
+                  else x;
+            in
+              makeDerivationExtensible mkDerivationSimple
+                (self: let super = rattrs self; in super // f self super))
+          args;
+    in public;
+
+  # Exactly makeDerivationExtensible (_: attrs), but specialized for performance
+  makeDerivationExtensible0 = mkDerivationSimple: attrs:
```

Comment:
Perhaps that name could be better in this case? What meaning does that `0` even have? It is yet another version of the function, but we aren't writing math here.

Reply:
Is `makeDerivationExtensibleConst` better?

It kind of is math, tbh, but that's no reason to adopt a worse syntax. Updating the comment though to use an equality operator rather than "exactly".